syntax = "proto2";

package maid.proto;

message ControllerProto
{
    //optional string service_name = 1;
    optional string method_name = 2;
    optional uint64 transmit_id = 3;
    optional bool stub = 4; // which side this message send
    optional bool is_canceled = 5; // not used currently
    optional bool failed = 6;
    optional string error_text = 7; // or error message
    optional bool notify = 9; // will not get an response
    optional bytes message = 10; // message

    optional string full_service_name = 11;
    //optional string full_method_name = 12;

    extensions 1000 to max;
}
